### PR TITLE
SyncBatchNorm.forward() to handle optional weight

### LINKED ISSUE #54495: SyncBatchNorm raises exception when affine=False
## 🐛 Bug

In nn.SyncBatchNorm:

There is a call to weight.contiguous() there:
https://github.com/pytorch/pytorch/blob/27048c1dfa80effabf17b8dca66cd2724dd502f8/torch/nn/modules/_functions.py#L12

But weight can be None if affine = False.
https://github.com/pytorch/pytorch/blob/27048c1dfa80effabf17b8dca66cd2724dd502f8/torch/nn/modules/batchnorm.py#L44 

So it crashes.

Bug was introduced in:
https://github.com/pytorch/pytorch/commit/d30f4d1dfd5237d89834363ce2cff9de4ee92811

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @albanD @mruberry

### PR DESCRIPTION
**Pull request by malfet**
Fixes #54495

### CODE REVIEW

**Comment by malfet** [comment]
TODO: Add test

**Review by jbschlosser — APPROVED** [review]
lgtm! agree with the todo for adding the test

**Review by xwang233 — APPROVED** [review]
Thanks for the fix!

**Comment by malfet** [comment]
@gchanan yeah, that's the reason why this PR has not been landed yet.

File: torch/testing/_internal/distributed/distributed_test.py [hunk 1]
```diff
@@ -3325,7 +3326,8 @@ def _test_DistributedDataParallel_SyncBatchNorm(self, gpu_subset, rank, local_bs
                 global_bs,
                 True,
                 offset,
-                dist.get_world_size()
+                dist.get_world_size(),
+                5 if affine is True else 2,
```
Comment (rohan-varma) [inline torch/testing/_internal/distributed/distributed_test.py:3330]: nit: just `5 if affine else 2`?
